### PR TITLE
change goog.require to provide empty interface by default.

### PR DESCRIPTION
**Pull request by rkirov**

### CODE REVIEW

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java [hunk 1]
```diff
@@ -329,6 +328,14 @@ public int compare(TypedVar o1, TypedVar o2) {
         if (desiredSymbols.contains(otherName) && other.getType() != null
             && !other.getType().isFunctionPrototypeType()
             && !isPrototypeMethod(other)) {
+          // For safety we need to special case goog.require to return the empty interface by default
+          // For existing namespaces we emit a goog.require string override that has the proper type.
```
Comment (mprobst) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: maybe link `See [place where we emit that].`?

File: src/resources/closure.lib.d.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ declare namespace ಠ_ಠ.clutz_internal {
   var GlobalError: ErrorConstructor;
   /** Represents a Closure type that is private, represented by an empty interface. */
   interface PrivateType {}
+  interface ClosureSymbolNotGoogProvided {}
```
Comment (mprobst) [inline src/resources/closure.lib.d.ts]: Add a JSDoc comment on what it means if you see this?

**Comment by mprobst** [comment]
LGTM.

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java (hunk 1 above)
Comment (rkirov) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: done

File: src/resources/closure.lib.d.ts (hunk 2 above)
Comment (rkirov) [inline src/resources/closure.lib.d.ts]: done

**Pull request merged**